src/server/pkg/sync/sync.go

```diff
@@ -4,16 +4,31 @@ package sync
 import (
 	"os"
 	"path/filepath"
+	"sync"
 	"syscall"
 
 	pachclient "github.com/pachyderm/pachyderm/src/client"
 	"github.com/pachyderm/pachyderm/src/client/pfs"
 	"github.com/pachyderm/pachyderm/src/server/pkg/obj"
 
-	log "github.com/Sirupsen/logrus"
 	"golang.org/x/sync/errgroup"
 )
 
+// Puller as a struct for managing a Pull operation.
+type Puller struct {
```

Comment:
1. I don't think makes sense, the `Mutex` isn't meant to be accessible publicly, it's just meant to be used to protect access to the `pipes` field in methods.

2. Definitely, will do.

3. The space argument makes sense, I'm inclined to leave it though since it does make setting values a little more verbose and we also use